### PR TITLE
Introduce initial Bazel build

### PR DESCRIPTION
**Pull request by armandomontanez**
Introduces the beginnings of a Bazel build for the Raspberry Pi Pico. The vast majority of preexisting build system configuration options are missing, but the build produces working binaries when used from an external project.

### CODE REVIEW

File: MODULE.bazel
Comment (armandomontanez) [inline MODULE.bazel:17]: I have an open commit that I need to include, so I'll be updating this shortly. If needed, I can get a full release of `rules_cc` cut before merging this PR.

**Comment by armandomontanez** [comment]
@kilograham I've put together an initial PR for adding a Bazel build, I have a few minor bits of cleanup before this is fully ready to merge, but I wanted to raise it to you now so you can start sharing any feedback/questions/concerns you may have now. @tpudlik has offered to review this from the Bazel side of things.

File: src/boards/BUILD.bazel
Comment (tpudlik) [inline src/boards/BUILD.bazel:101]: I would maybe call this `boards`. That's a special target name in that `@pico-sdk//src/boards:boards` can be abbreviated to just `@pico-sdk//src/boards` in lists of deps, etc. So it's weird to reserve this "most public" name for a target that's private and intended to be burned down!

Unless we want to hide the alias behind a `label_flag`, to allow SDK users to provide their own board definitions.

File: bazel/platform/BUILD.bazel
Comment (tpudlik) [inline bazel/platform/BUILD.bazel:1]: Should we define a host platform (even if building for it doesn't work yet)? It's a little weird that we have a host `constraint_value` that appears in a bunch of `select`s, but not host platform that selects that constraint value!

The host platform should include `HOST_CONSTRAINTS` from `@local_config_platform//:constraints.bzl` (a neat trick I only learned recently).

File: src/common/pico_base/BUILD.bazel
Comment (tpudlik) [inline src/common/pico_base/BUILD.bazel:49]: Could the comment explain _why_ we need to be careful? What should the maintainers be carefully watching out for?

File: src/rp2_common/boot_stage2/BUILD.bazel
Comment (tpudlik) [inline src/rp2_common/boot_stage2/BUILD.bazel:19]: Shouldn't `asminclude` go in here instead of line 28?

File: src/rp2_common/boot_stage2/BUILD.bazel
Comment (tpudlik) [inline src/rp2_common/boot_stage2/BUILD.bazel:51]: Consider using `copy_file` from Skylib, which has a simpler, cross-platform (Windows-friendly) interface: https://github.com/bazelbuild/bazel-skylib/blob/main/docs/copy_file_doc.md

File: src/rp2_common/boot_stage2/BUILD.bazel
Comment (tpudlik) [inline src/rp2_common/boot_stage2/BUILD.bazel:66]: Consider using `run_binary` from Skylib, which doesn't rely on the shell (https://github.com/bazelbuild/bazel-skylib/blob/main/docs/run_binary_doc.md). From [the docs](https://bazel.build/reference/be/general#genrule):

> Note that genrule requires a shell to interpret the command argument. It is also easy to reference arbitrary programs available on the PATH, however this makes the command non-hermetic and may not be reproducible. If you only need to run a single tool, consider using [run_binary](https://github.com/bazelbuild/bazel-skylib/blob/main/docs/run_binary_doc.md) instead.

File: src/rp2_common/pico_async_context/BUILD.bazel
Comment (tpudlik) [inline src/rp2_common/pico_async_context/BUILD.bazel:19]: Will all of these come from FreeRTOS? Another reason to upstream our FreeRTOS build file (https://pwbug.dev/326625641)!

File: src/rp2_common/pico_printf/BUILD.bazel
Comment (tpudlik) [inline src/rp2_common/pico_printf/BUILD.bazel:23]: Could you expand the TODO to mention why that's nontrivial?

File: bazel/README.md
Comment (armandomontanez) [reply bazel/README.md:60]: Added instructions for these. They'll get totally reworked shortly, but it's good to document what's there!

File: bazel/parse_version.py
Comment (armandomontanez) [reply bazel/parse_version.py]: Not relevant anymore.

File: src/rp2_common/boot_stage2/BUILD.bazel
Comment (armandomontanez) [reply src/rp2_common/boot_stage2/BUILD.bazel]: Yep, fixed.

File: src/rp2_common/pico_async_context/BUILD.bazel
Comment (armandomontanez) [reply src/rp2_common/pico_async_context/BUILD.bazel:19]: Yeah, these are all provided by FreeRTOS. Interestingly I'm not seeing CMake configuration options for turning this on/off despite it seeming to be an optional feature. I'll mark this on my TODO list.

**Review by armandomontanez — COMMENTED** [review]
Ready for another pass! :)

**Review by kilograham — APPROVED** [review]
I trust that this has stabilized enough for a baseline build... since it is entirely orthogonal to the CMake build, it can evolve independently.

**Pull request merged**
